extensions/amp-analytics/0.1/linker-reader.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getKeyValuePairs} from './linker';
+import {user} from '../../../src/log';
+import {removeParamsFromSearch} from '../../../src/url';
+import {getService, registerServiceBuilder} from '../../../src/service';
+import {hasOwn} from '../../../src/utils/object';
+import {parseUrlDeprecated, parseQueryString} from '../../../src/url';
+
+
+class LinkerReader {
+
+  constructor(window) {
+    this.win_ = window;
+
+    this.linkerParams_ = {};
+
+  }
+
+  /**
+   * Get the LINKER_PARAM(name, id) value from url and clean the value
+   * @param {string} name
+   * @param {string} id
+   * @return {?string}
+   */
+  get(name, id) {
+    if (!hasOwn(this.linkerParams_, name)) {
+      this.linkerParams_[name] = this.parseAndCleanQueryString_(name);
+    }
+
+    if (this.linkerParams_[name] && this.linkerParams_[name][id]) {
+      // Return the id value and remove the id from the object
+      const value = this.linkerParams_[name][id];
+      console.log('value is ', value);
+      delete this.linkerParams_[name][id];
```

Comment:
why delete?

Reply:
To avoid set cookie twice. I can either do `this.linkerParams_[name][id] = ''`, or simply delete it.